static/src/javascripts/lib/cmp.js

```diff
@@ -0,0 +1,89 @@
+// @flow
+import {
+    getAdConsentState,
+    thirdPartyTrackingAdConsent,
+} from 'common/modules/commercial/ad-prefs.lib';
+
+type PurposeEvent = 'functional' | 'performance' | 'advertisement';
+
+type PurposeCallback = (state: boolean | null) => void;
+
+type Purpose = {
+    state: boolean | null,
+    callbacks: Array<PurposeCallback>,
+};
+
+let cmpIsReady = false;
+
+const purposes: { [PurposeEvent]: Purpose } = {
+    functional: {
+        state: null,
+        callbacks: [],
+    },
+    performance: {
+        state: null,
+        callbacks: [],
+    },
+    advertisement: {
+        state: null,
```

Comment:
is boolean the correct type here? Do we need to be able to drill down into the IAB purposes?

Reply:
I'm not sure what you mean here. These are the general PECR purposes, which can be boolean or null.

Reply:
We're collecting `advertisement` via 5 questions, but as a boolean we're only able to represent a single answer. If we're intending to map those 5 answers to a single advertising answer them this is correct, as long as we're sure there are no use cases beyond the TCF that need access to the fine-grained details.

It's very likely correct as-is, but I wanted to draw attention to it so we can make sure.

Reply:
That's a decision that hasn't been made yet but the first option you mentioned appears to be what we're leaning towards so we opted to go ahead with this implementation to avoid getting blocked. So far no use cases came up that need fine-grained details. If any do come up we'll need to rethink this approach.